integration/oasis/chargingstrategy/chargingstrategy.go

```diff
@@ -11,24 +15,85 @@ func NewFakeChargingStrategyCreator(maxEnergyLevel float64) *fakeChargingStrateg
 	}
 }
 
+// @todo:
+// - Influence of returning candidate with no charge time and additional energy
 // CreateChargingStrategies returns different charging strategy
-// Initial implementation: 1 hour charge for 60% of max energy,
-//                         2 hour charge for 80%
-//                         4 hour charge for 100%
-func (f *fakeChargingStrategyCreator) CreateChargingStrategies() []ChargingStrategy {
-
-	return []ChargingStrategy{
-		ChargingStrategy{
-			ChargingTime:   3600,
+func (f *fakeChargingStrategyCreator) CreateChargingStrategies() []ChargingStatus {
+	return []ChargingStatus{
+		ChargingStatus{
 			ChargingEnergy: f.maxEnergyLevel * 0.6,
 		},
-		ChargingStrategy{
-			ChargingTime:   7200,
+		ChargingStatus{
 			ChargingEnergy: f.maxEnergyLevel * 0.8,
 		},
-		ChargingStrategy{
-			ChargingTime:   14400,
+		ChargingStatus{
 			ChargingEnergy: f.maxEnergyLevel,
 		},
 	}
 }
+
+// Fake charge strategy
+// From empty energy:
+//                    1 hour charge to 60% of max energy
+//                    2 hour charge to 80%, means from 60% ~ 80% need 1 hour
+//                    4 hour charge to 100%, means from 80% ~ 100% need 2 hours
+func (f *fakeChargingStrategyCreator) EvaluateCost(arrivalEnergy float64, targetState ChargingStatus) ChargingCost {
```

Comment:
`targetState` or `targetStatus`, which one would be better?

Reply:
Keep them consistency is most important, will keep `targetState`